### PR TITLE
Print recognized file extensions in resource builder error

### LINKED ISSUE #13450: Incorrect config filename yields confusing/unhelpful error message.
If I name my .yaml file as foo.yaml.2b , I get the following unhelpful error message:

  $ kubectl create -f example-pod.yaml.2b
  error: you must provide one or more resources by argument or filename

--v=9 offers no extra help

### PR DESCRIPTION
**Pull request by j3ffml**
closes #13450

### CODE REVIEW

**Comment by jlowdermilk** [comment]
@kubernetes/kubectl

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by kargakis** [comment]
LGTM

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ import (
 	"k8s.io/kubernetes/pkg/util/errors"
 )
 
+var schemaFileExtensions = []string{".json", ".stdin", ".yaml", ".yml"}
```
Comment (deads2k) [inline pkg/kubectl/resource/builder.go]: I would have expected this list to be shared with the bit creates content assist for the `--filenames` flag in the commands.

Comment (jlowdermilk) [inline pkg/kubectl/resource/builder.go]: Sorry, I don't know what bit you are referring to. Link to code?

Comment (deads2k) [inline pkg/kubectl/resource/builder.go]: Sorry, phone won't let me paste, but its in bash_comp_utils.go where the filename flag is registered.  It adds a hint for locating files with the right extension.

Comment (jlowdermilk) [inline pkg/kubectl/resource/builder.go]: Done.

**Comment by jlowdermilk** [comment]
@deads2k ptal. Added `.stdin` to filename extension completions, since the builder recognizes it.

**Comment by quinton-hoole** [comment]
e2e test failure was a flake due to #13534.  Retrying to try to induce auto-merge by merge-bot.

@k8s-bot test this again please

**Comment by piosz** [comment]
Removing LGTM since there were changes after it was given.

**Comment by deads2k** [comment]
lgtm

**Pull request merged**